### PR TITLE
Add Static Analysis config

### PR DESCRIPTION
**Pull request by connorhu**

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -0,0 +1,8 @@
+parameters:
+    level: 6
+    paths:
+        - src
+    phpVersion: 70200
```
Comment (derrabus) [inline phpstan.neon:5]:
```suggestion
    phpVersion: 80300
```

If you want to maintain your sanity, don't fix static analysis for PHP from the last decade.

File: .github/workflows/static-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - "master"
+    paths:
+      - .github/workflows/static-analysis.yml
+      - composer.*
+      - src/**
+      - phpstan*
+
+jobs:
+  coding-standards:
+    uses: "doctrine/.github/.github/workflows/static-analysis.yml@4.0.0"
```
Comment (derrabus) [inline .github/workflows/static-analysis.yml:23]: That workflow assumes that you have PHPStan _and_ Psalm installed.

File: phpstan.neon [hunk 3]
```diff
@@ -0,0 +1,8 @@
+parameters:
+    level: 6
```
[the remark this answers is not in the document]
Comment (connorhu) [reply phpstan.neon:2]: There are not an insane number of errors. I've started to fill in the missing phpdocs (DQL, @var etc) and that clears most of it out.

File: .github/workflows/static-analysis.yml (hunk 2 above)
Comment (connorhu) [reply .github/workflows/static-analysis.yml:23]: Yes, I found out. Now I'm thinking about a) getting the essential stuff out of there b) doing the psalm config as well.

File: phpstan.neon (hunk 3 above)
[the remark this answers is not in the document]
Comment (connorhu) [reply phpstan.neon:2]: What level do you recommend?

[the remark this answers is not in the document]
Comment (derrabus) [reply phpstan.neon:2]: One that gives us enough errors to fix before we move on to the next one.

File: .github/workflows/static-analysis.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      - "master"
+    paths:
+      - .github/workflows/static-analysis.yml
+      - composer.*
+      - src/**
+      - phpstan*
+      - psalm*
+
+jobs:
+  coding-standards:
```
Comment (greg0ire) [inline .github/workflows/static-analysis.yml:24]: Coding standards :thinking: ?

File: phpstan.neon (hunk 3 above)
[the remark this answers is not in the document]
Comment (connorhu) [reply phpstan.neon]: Then 3. I checked, that's the first level with 8 errors. I put them in the baseline.

File: .github/workflows/static-analysis.yml (hunk 4 above)
Comment (connorhu) [reply .github/workflows/static-analysis.yml]: Your eyes are like eagles'.

**Comment by connorhu** [comment]
Thank you for taking the time to review.

File: psalm.xml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+<psalm
+        errorLevel="2"
+        phpVersion="8.2"
```
Comment (greg0ire) [inline psalm.xml:4]: What's the rationale for choosing 8.2 here, and 8.3 for PHPStan?

Comment (connorhu) [reply psalm.xml:4]: To be honest, I have little experience of which version to run these tests on. I see on the ORM project that it is still on 8.2, @derrabus here suggested 8.3 for the stan, but I was careless with the psalm config. For doctrine1, for example, the lowest supported version is used (7.4). In my own projects, I try to choose the highest version. I'll upgrade to 8.3.

File: psalm.xml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+<psalm
+        errorLevel="2"
```
Comment (derrabus) [inline psalm.xml:3]: Same here. The level might be too low to start with.

Comment (connorhu) [reply psalm.xml]: I didn't know that 8 is the lowest. I was expecting similar functionality as phpstan. Updated.

**Review by derrabus — APPROVED** [review]

**Comment by connorhu** [comment]
Thank you for your review!

**Pull request merged**